Refactor SwapScene layout to use safeAreaView

DRadmir opened the pull request:
Replaces the ZStack-based layout with a safeAreaView wrapper for the bottomActionView in SwapScene. This change improves layout handling and ensures the bottom action view respects safe area insets.

Fix: https://github.com/gemwalletcom/gem-ios/issues/1169

## iOS 26
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 12 26" src="https://github.com/user-attachments/assets/107070a1-c4ca-4965-bfa9-d4fd64ee9c96" />
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 12 32" src="https://github.com/user-attachments/assets/78ab3935-b37e-4718-bcbf-f75ca2903510" />
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 12 42" src="https://github.com/user-attachments/assets/7232e690-d7a0-4963-8e5b-b8bc5b6ea24f" />
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 15 38" src="https://github.com/user-attachments/assets/a614c0f9-e722-4b11-bb67-775d75de4d24" />
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 15 42" src="https://github.com/user-attachments/assets/7a249907-921a-4bef-859f-8a280a355a51" />
<img width="240" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-18 at 14 15 55" src="https://github.com/user-attachments/assets/43462112-beb5-4be2-a4cf-0c50b33cae6f" />

## iOS 18.5
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 11 06" src="https://github.com/user-attachments/assets/130d14a7-57c7-42e0-9fcd-161452fe95a9" />
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 11 14" src="https://github.com/user-attachments/assets/184febd6-956b-43df-9eee-3a61feb5ea6f" />
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 11 36" src="https://github.com/user-attachments/assets/b6f8e7ce-dcdf-4176-92fb-1840e6e7fff4" />
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 11 44" src="https://github.com/user-attachments/assets/4c927c33-fbdf-4f40-b261-5b39e7b0b271" />
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 12 03" src="https://github.com/user-attachments/assets/0ca2efa0-7898-4033-b29e-c3dbd4f7b9d9" />
<img width="240" alt="Simulator Screenshot - iPhone 16e - 2025-09-18 at 14 12 15" src="https://github.com/user-attachments/assets/1fd95493-3e28-40ac-993b-6b61fbbfba1b" />

DRadmir committed:
Refactor SwapScene layout to use safeAreaView

Replaces the ZStack-based layout with a safeAreaView wrapper for the bottomActionView in SwapScene. This change improves layout handling and ensures the bottom action view respects safe area insets.

DRadmir committed:
Refactor SwapTokenView to handle optional model

Updated SwapTokenView to accept an optional model and render SwapTokenEmptyView when the model is nil. Simplified SwapScene by removing conditional logic for rendering SwapTokenView or SwapTokenEmptyView, delegating this responsibility to SwapTokenView. Also adjusted padding in SwapTokenView for improved UI consistency.

DRadmir committed:
Merge branch 'main' into 1169-not-able-to-see-swap-button-in-swap-screen

gemdev111 commented on Features/Swap/Sources/Views/SwapAssetView.swift:
```diff
@@ -45,7 +53,7 @@ struct SwapTokenView: View {
                     }
                 }
-                .padding(.vertical, 8)
+                .padding(.vertical, 4)
```
can we use Style for 2/4

gemcoder21 commented on Features/Swap/Sources/Views/SwapAssetView.swift:
```diff
@@ -7,14 +7,22 @@ import Primitives
 struct SwapTokenView: View {
     
-    let model: SwapTokenViewModel
+    let model: SwapTokenViewModel?
```
Can we not use optional? it should not handle `SwapTokenEmptyView` state in here

gemcoder21 commented on Features/Swap/Sources/Views/SwapAssetView.swift:
```diff
@@ -7,14 +7,22 @@ import Primitives
     var body: some View {
+        if let model {
+            inputView(model: model)
+        } else {
```
just move this out, scene will drive this logic

DRadmir committed:
Refactor SwapTokenView usage and handling of nil models

Updated SwapScene to explicitly handle cases where swapTokenModel returns nil by showing SwapTokenEmptyView. Refactored SwapTokenView to require a non-optional model and removed internal nil checks, simplifying the view logic.
